envoy_binary_test.sh

```diff
@@ -1,10 +0,0 @@
-#!/bin/bash
-#
-
-set -e
-
-# Just test that the binary was produced and can be executed.
-# envoy --help will give a success return code if working.
-cilium-envoy --help
-
-echo "PASS"
```

Comment:
ah I thought we changed to `cilium-envoy version`, so that we can check for the hash as well in CI. 

But yes, this file is not related to istio cleanup, will add it back.